lib/protoparser/influx/stream/streamparser.go

```diff
@@ -57,18 +58,21 @@ func Parse(r io.Reader, isGzipped bool, precision, db string, callback func(db s
 		tsMultiplier = -1e3 * 3600
 	}
 
-	ctx := getStreamContext(r)
-	defer putStreamContext(ctx)
-	for ctx.Read() {
+	ctx := getProcessingContext(r)
+	defer putProcessingContext(ctx)
+	for ctx.Read(isStreamMode) {
```

Comment:
Having this uniform will make batch processing to touch waitgroups and concurrency limiter more frequently that it should.
WDYT about having `ctx.Read()` and `ctx.ReadLine` instead? That should be easier to read and understand at both: here and in `ctx.Read` implementation.